gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureLocalPlugin.java

```diff
@@ -127,10 +129,39 @@ private void setupGenericConjureProjects(
                                         task.setSource(conjureIrConfiguration);
                                         task.setExecutablePath(extractConjureGeneratorTask::getExecutable);
                                         task.setOptions(() -> conjureExtension.getGenericOptions(subprojectName));
-                                        task.setOutputDirectory(subproject.file(subprojectName));
+
+                                        final String outputDir;
+                                        if (conjureExtension.getGenericOptions(subprojectName).has("outputDir")) {
```

Comment:
@ferozco this does not get evaluated lazily, i.e., I don't have access to options at this point in the code. any idea why? your recommendation for working around this?

Reply:
I would recommend to support Java as a first class language and wire it up seperately from the generic generators. I think this is desirable since you'll also will have to configure source sets/idea modules/dependencies for the java project